copylot/hardware/lasers/vortran/test/test_vortran.py

```diff
@@ -1,49 +1,160 @@
-# %%
-from copylot.hardware.lasers.vortran import vortran
-import time
-
-
-def demo_get_laser_list():
-    """
-    Get the Voltran Lasers Connected to the system
-    """
-    laser_list = vortran.VortranLaser.get_lasers()
-    print(laser_list)
-
-
-def demo_set_power():
-    """
-    Test Setting the power of the laser.
-    To read the value, emission = 1
-    """
-    power = 3.0
-    laser_list = vortran.VortranLaser.get_lasers()
-    laser = vortran.VortranLaser(port=laser_list[0][0])
-    laser.toggle_emission = 1
-    laser.laser_power = 2.0
-    assert (laser.laser_power - power) < (power * 0.2)
-    power = float(5.0)
-    laser.laser_power = power
-    assert (laser.laser_power - power) < (power * 0.2)
-    laser.toggle_emission = 0
-    laser.disconnect()
+import pytest
+from unittest import mock
```

Comment:
do we need this import from `unittest`?

Reply:
technically we don't because the `mocker` fixture provided by `pytest-mock` proxies the `unittest.mock` API. But the downside of always using the proxied API is that introspection is lost (at least in vs code) because intellisense does not know about pytest fixtures. So I used the 'real' `unittest.mock` API to instantiate standalone mocks (as these do not need lifecycle management) and `mocker.patch` for creating the patches (as these obviously need to/should be torn down at the end of each test).

Reply:
I don't use VSCode so haven't faced the proxy introspection situation but we can leave it with given reasoning. Thank you for the explanation.